src/core/tests/any.cpp

```diff
@@ -419,15 +419,15 @@ TEST_F(AnyTests, constructFromVariantImpl) {
 TEST_F(AnyTests, dynamicPointerCastToVariantWrapper) {
     OPENVINO_SUPPRESS_DEPRECATED_START
     Any p = std::make_shared<ngraph::VariantWrapper<std::string>>("42");
-    auto str_variant = std::dynamic_pointer_cast<ngraph::VariantWrapper<std::string>>(p);
+    auto str_variant = ov::Any(p).as<std::shared_ptr<ngraph::VariantWrapper<std::string>>>();
     ASSERT_EQ("42", str_variant->get());
```

Comment:
These test items are flagged with "OPENVINO_SUPPRESS_DEPRECATED_START", do we need keep them?

Reply:
I think we need to remove `VariantWrapper` which is deprecated and in this case we will be able to remove `OPENVINO_SUPPRESS_DEPRECATED_START`